posthog/models/filters/mixins/simplify.py

```diff
@@ -82,12 +83,12 @@ def _simplify_property_group(self, team: "Team", prop_group: "PropertyGroup", **
             if isinstance(group, PropertyGroup):
                 new_groups.append(self._simplify_property_group(team, group))
             elif isinstance(group, Property):
-                new_groups.extend(self._simplify_property(team, group))  # type: ignore
+                new_groups.append(self._simplify_property(team, group))
 
         prop_group.values = new_groups
         return prop_group
 
-    def _simplify_property(self, team: "Team", property: "Property", **kwargs) -> List["Property"]:
+    def _simplify_property(self, team: "Team", property: "Property", **kwargs) -> "PropertyGroup":
```

Comment:
@EDsCODE @rcmarron before I jump into fixing lots of existing tests as a consequence of this change, lmk if this makes sense?

Idea is to always pass around propertygroups now, which allows combining these in more flexible ways, and keeps maintaining the invariant that a single values list can't have both property and propertyGroups.

Reply:
Yup this should be fine